lib/ae_page_objects.rb

```diff
@@ -35,22 +32,80 @@ def browser
       end
     end
 
-    def wait_until(seconds_to_wait = nil, error_message = nil)
-      seconds_to_wait ||= default_max_wait_time
-      start_time      = Time.now
+    def wait_until(seconds_to_wait = nil, error_message = nil, &block)
```

Comment:
You've called this commit `pk - passing tests` but really this one contains the bulk of the changes.